Framework/MDAlgorithms/src/CompactMD.cpp

```diff
@@ -0,0 +1,140 @@
+#include "MantidMDAlgorithms/CompactMD.h"
+#include "MantidAPI/IMDIterator.h"
+using namespace Mantid::API;
+using namespace Mantid::Geometry;
```

Comment:
Can't see any progress reporting on this. I would suggest always putting Progress reporting on MD-algorithms due to the limiting behaviour O(n<sup>m</sup>) where *m* is the number of dimensions